saleor/graphql/core/types/taxes.py

```diff
@@ -389,13 +402,47 @@ def calculate_checkout_discounts(checkout_info):
             )
 
         def map_discounts(discounts):
-            return [
-                {"name": discount.name, "amount": discount.amount}
-                for discount in discounts
-                # Only order level discounts, like entire order vouchers,
-                # order promotions and manual discounts should be taken into account
-                if is_order_level_discount(discount)
-            ]
+            # Only order level discounts, like entire order vouchers,
+            # order promotions and manual discounts should be taken into account.
+            # Manual discount needs to be split into subtotal and shipping portions.
+            taxable_discounts = []
+            for discount in discounts:
+                if discount.type != DiscountType.MANUAL:
+                    continue
+                subtotal = root.subtotal.net
+                shipping = root.base_shipping_price
+                subtotal_discount, shipping_discount = split_manual_discount(
```

Comment:
`subtotal_discount` or `shipping_discount` can be == 0, in that case there is no need to add them to the response.